pkg/kritis/cryptolib/verifier.go

```diff
@@ -70,23 +81,32 @@ func NewVerifier(imageDigest string, publicKeySet []PublicKey) (Verifier, error)
 // VerifyAttestation verifies an Attestation. See Verifier for more details.
 func (v *verifier) VerifyAttestation(att *Attestation) error {
 	var (
-		err     error
-		payload []byte
+		err       error
+		payload   []byte
+		publicKey PublicKey
 	)
 
 	// Extract the public key from `publicKeySet` whose ID matches the one in
 	// `att`.
-	// TODO: Replace no-op with correct implementation.
-	publicKey := v.PublicKeySet[0]
+	foundKey := false
+	for _, key := range v.PublicKeySet {
```

Comment:
Consider changing the verifier struct so that PublicKeySet becomes a map:
id -> PublicKey
(And you'd have to change NewVerifier() to populate this map.)

If you do this, then you can avoid this loop and just do a map lookup.

Reply:
Done.